Scripts/aws_stack_provision.ps1

```diff
@@ -71,67 +155,115 @@ else{
-    		$RetryCount = 15
-
-    		while ( ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceStatus -ne "available") -and ($RetryCount -gt 0)){
-
-      			Start-Sleep -Seconds 120
-      			$RetryCount -= 1 
-      			if ( ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceStatus -eq "starting") -or ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceStatus -eq "backing-up")  ) {
-
-         			Write-Host "Oracle RDS is getting started"}}
-
-		if ( $RetryCount -le 0 ) {
-        		throw "Timeout: 30 minutes expired waiting for RDS to be in availabe state" }
-    		Write-Host "Oracle RDS is in Available state"}
-
-  	else{
-    		throw "Oracle Database is in $ORACLE_DB_STATUS state"}}
-
-elseif ($ORACLE_DB_COUNT -eq 0){
-   	Write-Host "No existing oracle database exist with required lansa version"
-   	$ORACLE_SNAPSHOT_COUNT = ((Get-RDSDBSnapshot -DBInstanceIdentifier $ORACLE_DB_IDENTIFIER).DBSnapshotArn).count
-   	if ($ORACLE_SNAPSHOT_COUNT -eq 1){
-     		Write-Host "Found 1 snapshot for database identifier $ORACLE_DB_IDENTIFIER"
-     		#New-CNFStack ......................................................
-
-     		$RetryCount = 15
-     		while ( ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceStatus -ne "available") -and ($RetryCount -gt 0)){
-
-      			Start-Sleep -Seconds 120
-      			$RetryCount -= 1
-      			if ( ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceStatus -eq "starting") -or ((Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceStatus -eq "backing-up")  ) {
-
-         			Write-Host "Oracle RDS is getting started"}}
-
-     		if ( $RetryCount -le 0 ) {
-        		throw "Timeout: 30 minutes expired waiting for RDS to be in availabe state" }}
-
-   	elseif ($ORACLE_SNAPSHOT_COUNT -eq 0){
-     		throw "Found 0 snapshot for database identifier $ORACLE_DB_IDENTIFIER"}
-   	else{
-     		throw "Found more than 1 snapshot for database identifier $ORACLE_DB_IDENTIFIER"}}
+if ($ORACLE_DB_COUNT -eq 1)
+{
+
+   Write-Host "Oracle RDS with Lansa Version tag = $lansa_version already exist"
+   $ORACLE_DB_STATUS = (Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceStatus
+   $ORACLE_DB_ARN = (Get-RDSDBInstance -Filter @{Name="db-instance-id"; Values=$ORACLE_DB_IDENTIFIER}).DBInstanceArn
+
+   if ($ORACLE_DB_STATUS -eq "available")
+   {
+      Write-Host "Oracle RDS is in $ORACLE_DB_STATUS state and its ARN is $ORACLE_DB_ARN"
+   }
+
+   elseif ($ORACLE_DB_STATUS -eq "stopped")
+   {
+      Write-Host "Oracle RDS is in $ORACLE_DB_STATUS state and its ARN is $ORACLE_DB_ARN"
+      Start-RDSDBInstance -DBInstanceIdentifier $ORACLE_DB_IDENTIFIER
+      Write-Host "Waiting for Oracle RDS to be in Available state"
+
+      $RetryCount = 15
```

Comment:
This code is duplicated at line 235 onwards. Make a Function of it.

Same with the MYSQL code.

Any code that you are duplicating, make a Function of it. Thats all I will accept. Do it right the first time.

Reply:
I will create function for it.